homeassistant/components/sensor/child_sensor.py

```diff
@@ -0,0 +1,144 @@
+"""
```

Comment:
I'm also not sure I'm convinced as to why these need to be so generic. It would be a very difficult task to make a child sensor and switch that is appropriately generic for all use cases. That is basically what the ABCs are supposed to do while still allowing the developer full customization of the entity. These could probably be made more Vera specific without losing much. Again, though, definitely up for discussion.